packages/loader/container-loader/src/protocol.ts

```diff
@@ -99,3 +99,24 @@ export class ProtocolHandler extends ProtocolOpHandler implements IProtocolHandl
 		}
 	}
 }
+
+/**
+ * Function to check whether the protocol handler should process the Signal.
+ * The protocol handler should strictly handle only ClientJoin, ClientLeave
+ * and Clear signal types.
+ */
+export function protocolHandlerShouldProcessSignal(message: ISignalMessage) {
+	// Signal originates from server
+	if (message.clientId === null) {
+		const innerContent = message.content as { content: unknown; type: string };
```

Comment:
Maybe consider if there's a name we can put on this type.  Really this is implying that ISignalMessage is incorrectly typed, because we think we know more about its contents than the typing implies.  Not blocking though.

Reply:
I'll put this in another PR :)